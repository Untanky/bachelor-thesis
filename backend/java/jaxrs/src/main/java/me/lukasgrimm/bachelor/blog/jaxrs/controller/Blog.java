package me.lukasgrimm.bachelor.blog.jaxrs.controller;

import me.lukasgrimm.bachelor.dao.PostDAO;
import me.lukasgrimm.bachelor.models.Post;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.List;

@Path("/blog")
public class Blog {

    PostDAO dao;

    @Inject
    Blog(PostDAO dao) {
        this.dao = dao;
    }

    @GET
    @Path("/post")
    @Produces(MediaType.APPLICATION_JSON)
    public Response fetchAllPosts() {
        return null;
    }

    @POST
    @Path("/post")
    @Consumes(MediaType.APPLICATION_JSON)
    public Response createPost(Post post) {
        return null;
    }

    @PUT
    @Path("/post/{postId}")
    @Consumes(MediaType.APPLICATION_JSON)
    public Response updatePost(@PathParam("postId") int postId, Post post) {
        return null;
    }

    @PUT
    @Path("/post/{postId}")
    public Response deletePost(@PathParam("postId") int postId) {
        return null;
    }
}
