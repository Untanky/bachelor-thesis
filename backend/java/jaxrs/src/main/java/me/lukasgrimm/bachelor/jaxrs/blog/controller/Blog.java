package me.lukasgrimm.bachelor.jaxrs.blog.controller;

import me.lukasgrimm.bachelor.models.Post;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.xml.ws.Response;
import java.util.List;

@Path("/blog")
public class Blog {

    @GET
    @Path("/post")
    @Produces(MediaType.APPLICATION_JSON)
    public Response<List<Post>> fetchAllPosts() {
        return null;
    }

    @POST
    @Path("/post")
    @Consumes(MediaType.APPLICATION_JSON)
    public Response<Void> createPost(Post post) {
        return null;
    }

    @PUT
    @Path("/post/{postId}")
    @Consumes(MediaType.APPLICATION_JSON)
    public Response<Void> updatePost(@PathParam("postId") int postId, Post post) {
        return null;
    }

    @PUT
    @Path("/post/{postId}")
    public Response<Void> deletePost(@PathParam("postId") int postId) {
        return null;
    }
}
