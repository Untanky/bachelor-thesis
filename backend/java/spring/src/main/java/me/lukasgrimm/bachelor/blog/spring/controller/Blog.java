package me.lukasgrimm.bachelor.blog.spring.controller;

import me.lukasgrimm.bachelor.dao.PostDAO;
import me.lukasgrimm.bachelor.models.Post;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/blog")
@CrossOrigin(origins = "*")
public class Blog {

    PostDAO dao;

    @Autowired
    public Blog(PostDAO dao) {
        this.dao = dao;
    }

    @GetMapping("/post")
    List<Post> fetchAllPosts() {
        return dao.findAll();
    }

    @PostMapping("/post")
    ResponseEntity<Void> createPost(@RequestBody Post post) {
        try {
            dao.create(post);
        } catch (IllegalArgumentException e) {
            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
        }

        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
    }

    @PutMapping("/post/{postId}")
    ResponseEntity<Void> updatePost(@PathVariable("postId") int postId, @RequestBody Post post) {

        if (postId != post.getId()) {
            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
        }

        try {
            dao.update(post);
        } catch (IllegalArgumentException e) {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }

        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
    }

    @DeleteMapping("/post/{postId}")
    ResponseEntity<Void> deletePost(@PathVariable("postId") int postId) {

        try {
            dao.delete(postId);
        } catch (IllegalArgumentException e) {
            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
        }

        return new ResponseEntity<>(HttpStatus.NO_CONTENT);
    }
}
