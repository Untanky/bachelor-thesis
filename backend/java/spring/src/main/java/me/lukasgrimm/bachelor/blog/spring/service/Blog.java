package me.lukasgrimm.bachelor.blog.spring.service;

import me.lukasgrimm.bachelor.dao.PostDAO;
import me.lukasgrimm.bachelor.dao.PostDBDAO;
import me.lukasgrimm.bachelor.models.Post;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
public class Blog {

    @Autowired
    PostDAO dao;

    @GetMapping("/")
    List<Post> fetchAllPosts() {

    }

    @PostMapping("/")
    void createPost(@RequestBody Post post) {

    }

    @PutMapping("/{postId}")
    void updatePost(@PathVariable("postId") int postId, @RequestBody Post post) {

    }

    @DeleteMapping("/{postId}")
    void deletePost(@PathVariable("postId") int post) {

    }
}
