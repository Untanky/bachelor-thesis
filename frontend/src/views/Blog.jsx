import React from 'react';

import PostList from '../components/PostList';

import './Blog.scss';

const Blog = () => {
  const postList = [
    {
      id: '1',
      title: 'Test title',
      description: 'This is the description for the test post. It contains vital data for the post and puts it into context.',
      numberOfComments: 6,
      url: 'post/1'
    },
    {
      id: '2',
      title: 'My newest post',
      description: 'This is the description for the test post. It contains vital data for the post and puts it into context.',
      numberOfComments: 0,
      url: 'post/2'
    },
    {
      id: '3',
      title: 'Post with some content',
      description: 'This is the description for the test post. It contains vital data for the post and puts it into context.',
      numberOfComments: 200,
      url: 'post/3'
    },
  ];

  return (
    <div className="blog">
      <PostList posts={postList} />
    </div>
  )
};

export default Blog;
