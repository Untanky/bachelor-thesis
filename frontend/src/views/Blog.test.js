import React from 'react';
import Blog from './Blog';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <Blog />
      </StaticRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
