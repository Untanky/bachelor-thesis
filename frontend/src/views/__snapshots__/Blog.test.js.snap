// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="blog"
>
  <div
    className="post-list"
  >
    <div
      className="post"
    >
      <h2
        className="post-title"
      >
        Test title
      </h2>
      <p
        className="post-description"
      >
        This is the description for the test post. It contains vital data for the post and puts it into context.
      </p>
      <a
        className="button-link"
        href="/post/edit/1"
        onClick={[Function]}
      >
        Edit post
      </a>
    </div>
    <div
      className="post"
    >
      <h2
        className="post-title"
      >
        My newest post
      </h2>
      <p
        className="post-description"
      >
        This is the description for the test post. It contains vital data for the post and puts it into context.
      </p>
      <a
        className="button-link"
        href="/post/edit/2"
        onClick={[Function]}
      >
        Edit post
      </a>
    </div>
    <div
      className="post"
    >
      <h2
        className="post-title"
      >
        Post with some content
      </h2>
      <p
        className="post-description"
      >
        This is the description for the test post. It contains vital data for the post and puts it into context.
      </p>
      <a
        className="button-link"
        href="/post/edit/3"
        onClick={[Function]}
      >
        Edit post
      </a>
    </div>
  </div>
</div>
`;
